src/node.js

```diff
@@ -206,6 +207,13 @@ class Node {
             this.api.apiHost = addr.address
             this.api.apiPort = addr.port
 
+            if (gwmatch) {
+              this.gatewayAddr = gwmatch[2]
+              const addr = multiaddr(this.gatewayAddr).nodeAddress()
```

Comment:
How about exposing `gatewayAddr` and `apiAddr` as the main thing, and making the others simple helper functions which just depend on those?